app/components/Notification/index.jsx

```diff
@@ -21,17 +21,26 @@ let NotifiItem = React.createClass({
         sender: React.PropTypes.object.isRequired,
         time: React.PropTypes.string.isRequired,
         message: React.PropTypes.string.isRequired,
+        url: React.PropTypes.string,
         read: React.PropTypes.bool
     },
     contextTypes: {
-        muiTheme: React.PropTypes.object
+        muiTheme: React.PropTypes.object,
+        router: React.PropTypes.func.isRequired
+    },
+
+    _transitionToPost(){
+        var parser = document.createElement('a');
+        parser.href = this.props.url;
+        this.context.router.transitionTo(parser.pathname);
```

Comment:
What's the problem of just using `this.context.router.transitionTo(this.props.url);` ?